### ISSUE TITLE
Healthcheck enabled containers stuck in initializing state after they move to another host

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#12140]
**Rancher versions:**
rancher/server: v1.6.15-rc7

**Steps to Reproduce:**
1. Create a setup with 3 hosts "h1, h2, h3" on AWS
2. Launch EBS catalog and create a ebs volume "testvol" from the UI storage tab
3. Create healthcheck enabled service "service1" using ebs volume "testvol"
4. Power OFF the host "h1" where service1 exists.

**Results:**
After healthcheck enabled containers move to another host, they are stuck in initializing state.

Healthcheck on the other host reports:
```
3/16/2018 11:45:13 AMtime="2018-03-16T18:45:13Z" level=error msg="Failed to report status cb2ac09f-cfff-4ff7-b706-7f5e80c60a46_6d8d41d1-0cb7-423b-a02f-3fa45dba4c04_1=INIT: Bad response from [http://35.202.26.149:8080/v1/serviceevents], statusCode [409]. Status [409 Conflict]. Body: [{\"id\":\"c4c4e63b-540d-4cf3-9985-bb34a791507a\",\"type\":\"error\",\"links\":{},\"actions\":{},\"status\":409,\"code\":\"Conflict\",\"message\":\"Conflict\",\"detail\":null,\"baseType\":\"error\"}]"
3/16/2018 11:45:13 AMtime="2018-03-16T18:45:13Z" level=error msg="Failed to report status cb2ac09f-cfff-4ff7-b706-7f5e80c60a46_4671f320-d22a-4d1f-8d4a-98367460e456_1=INIT: Bad response from [http://35.202.26.149:8080/v1/serviceevents], statusCode [409]. Status [409 Conflict]. Body: [{\"id\":\"130086be-7887-4c04-96d1-77130c73b08d\",\"type\":\"error\",\"links\":{},\"actions\":{},\"status\":409,\"code\":\"Conflict\",\"message\":\"Conflict\",\"detail\":null,\"baseType\":\"error\"}]"
3/16/2018 11:45:15 AMtime="2018-03-16T18:45:15Z" level=error msg="Failed to report status cb2ac09f-cfff-4ff7-b706-7f5e80c60a46_6d8d41d1-0cb7-423b-a02f-3fa45dba4c04_1=INIT: Bad response from [http://35.202.26.149:8080/v1/serviceevents], statusCode [409]. Status [409 Conflict]. Body: [{\"id\":\"116790f8-c560-4c1d-9c5c-299a5d963f6e\",\"type\":\"error\",\"links\":{},\"actions\":{},\"status\":409,\"code\":\"Conflict\",\"message\":\"Conflict\",\"detail\":null,\"baseType\":\"error\"}]"
3/16/2018 11:45:15 AMtime="2018-03-16T18:45:15Z" level=error msg="Failed to report status cb2ac09f-cfff-4ff7-b706-7f5e80c60a46_4671f320-d22a-4d1f-8d4a-98367460e456_1=INIT: Bad response from [http://35.202.26.149:8080/v1/serviceevents], statusCode [409]. Status [409 Conflict]. Body: [{\"id\":\"c2a885aa-10bd-4572-9103-1f8184697e7d\",\"type\":\"error\",\"links\":{},\"actions\":{},\"status\":409,\"code\":\"Conflict\",\"message\":\"Conflict\",\"detail\":null,\"baseType\":\"error\"}]"

```

**Comment by kinarashah** [comment]
It looks like there's some race condition, if you just edit the ipsec service (for instance, change the description), health state update will be triggered and it should be fine.

**Comment by schuylr** [comment]
I saw this everywhere over the weekend on my Rancher environment, v1.6.14 (non-RC). It caused many of my services to die and had a catastrophic failure on all of my websites.

The only way for me to fix this was by deleting the affected service and cloning a new one with the same health checks, which seem to assign them to healthy health check containers. Rancher does not seem to re-allocate health checks on the fly when the responsible hosts have dead health check containers.

In addition, editing or restarting the IPSec service didn't help.

**Comment by smalenfant** [comment]
I created a new cluster yesterday and I'm getting those after I create the second node. No services installed, just adding nodes in the cluster. IPSEC is also failing after the second node addition.

I'm using RancherOS 1.3.0 and Rancher 1.6.16 build on top of KVM. Master has a public IP and nodes have private IPs (10.x.x.x). There is firewall in between.

**Comment by nagliyvred** [comment]
We are seeing the same behaviour in our production cluster every day: whenever a node is added healthcheck goes crazy and the only solution is to delete ipsec containers one-by-one and then delete healthcheck containers one-by-one so that they re-created from scratch...

Are there any other workarounds? This is causing us major disruptions in prod

**Comment by pkaramol** [comment]
I am having the same problem.
Working on AWS and when adding a second instance as Rancher host manually, (i.e. not via the GUI which is extremely cumbersome either way) I am getting the above error.

**Comment by nagliyvred** [comment]
Any news or at least pointers on what this could be and/or how this could be worked around? 
This is being a major issue for us in production and the last stone that will make us move away from rancher. 
I found that sometimes after launching a new instance, if I restart rancher-agent it might fade away. Could it be that the agent somehow gets bound to the wrong ip address?

**Comment by deniseschannon** [comment]
@nagliyvred there is a workaround based on this comment: https://github.com/rancher/rancher/issues/12140#issuecomment-373829233
what version of Rancher are you using?

@soumyalj after talking to @alena1108, there were some changes that indicate that it should be fixed i nthe last release v1.6.18.

**Comment by soumyalj** [comment]
Tested with v1.6.19-rc4. The use case was tested several times and could not be reproduced. 
https://github.com/rancher/rancher/issues/14484
